app/services/workflow_update_service.rb

```diff
@@ -1,15 +1,17 @@
+require_relative 'mixins/group_validate_mixin'
+
 class WorkflowUpdateService
+  include GroupValidateMixin
   attr_accessor :workflow
 
   def initialize(workflow_id)
     self.workflow = Workflow.find(workflow_id)
   end
 
   def update(options)
-    options[:access_control_entries] =
-      (options[:group_refs] || []).collect do |uuid|
-        AccessControlEntry.new(:group_uuid => uuid, :permission => 'approve')
-      end
+    if options[:group_refs]
+      options[:group_refs] = validate_approver_groups(options[:group_refs])
```

Comment:
Should we call `validate_and_update_approver_groups` here? so don't need line 16 anymore

Reply:
no. the workflow is saved in the next line. We don't need to save multiple times.